pinot-core/src/main/java/org/apache/pinot/core/transport/grpc/GrpcQueryClient.java

```diff
@@ -16,20 +16,26 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.pinot.server.starter.grpc;
+package org.apache.pinot.core.transport.grpc;
 
-import io.grpc.stub.StreamObserver;
+import io.grpc.Channel;
+import io.grpc.ManagedChannelBuilder;
+import java.util.Iterator;
 import org.apache.pinot.common.proto.PinotQueryServerGrpc;
 import org.apache.pinot.common.proto.Server;
 
-/**
- * Handler for grpc server requests.
- * As data becomes available server responses will be added to the result stream.
- * Once the request is complete the client will aggregate the result metadata.
- */
-public class PinotQueryHandler extends PinotQueryServerGrpc.PinotQueryServerImplBase {
-  @Override
-  public void submit(Server.ServerRequest request, StreamObserver<Server.ServerResponse> responseObserver) {
 
+public class GrpcQueryClient {
```

Comment:
shall we consider move this to pinot-common? So other system integration won't require pinot-core as the dependency

Reply:
Moved to `pinot-common`